fe/fe-core/src/main/java/org/apache/doris/catalog/StoragePolicyResource.java

```diff
@@ -0,0 +1,128 @@
+
+package org.apache.doris.catalog;
+
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.proc.BaseProcResult;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Map;
+
+/**
+ * Policy resource for olap table
+ *
+ * Syntax:
+ * CREATE RESOURCE "storage_policy_name"
+ * PROPERTIES(
+ *      "type"="storage_policy",
+ *      "cooldown_datetime" = "2022-06-01", // time when data is transfter to medium
+ *      "cooldown_ttl" = "1h"， // data is transfter to medium after 1 hour
+ *      "s3_resource" = "my_s3" // point to a s3 resource
+ * );
+ */
+public class StoragePolicyResource extends Resource {
+    // required
+    private static final String STORAGE_RESOURCE = "storage_resource";
+    // optional
+    private static final String COOLDOWN_DATETIME = "cooldown_datetime";
+    private static final String COOLDOWN_TTL = "cooldown_ttl";
+
+    private static final String DEFAULT_COOLDOWN_DATETIME = "9999-01-01 00:00:00";
+    private static final String DEFAULT_COOLDOWN_TTL = "1h";
+
+    @SerializedName(value = "properties")
+    private Map<String, String> properties;
+
+    public StoragePolicyResource(String name) {
+        this(name, Maps.newHashMap());
+    }
+
+    public StoragePolicyResource(String name, Map<String, String> properties) {
+        super(name, ResourceType.STORAGE_POLICY);
+        this.properties = properties;
+    }
+
+    public String getProperty(String propertyKey) {
+        return properties.get(propertyKey);
+    }
+
+    @Override
+    protected void setProperties(Map<String, String> properties) throws DdlException {
+        Preconditions.checkState(properties != null);
+        this.properties = properties;
+        // check properties
+        // required
+        checkRequiredProperty(STORAGE_RESOURCE);
+        // optional
+        checkOptionalProperty(COOLDOWN_DATETIME, DEFAULT_COOLDOWN_DATETIME);
+        checkOptionalProperty(COOLDOWN_TTL, DEFAULT_COOLDOWN_TTL);
+    }
+
+    private void checkRequiredProperty(String propertyKey) throws DdlException {
+        String value = properties.get(propertyKey);
+
+        if (Strings.isNullOrEmpty(value)) {
+            throw new DdlException("Missing [" + propertyKey + "] in properties.");
+        }
+    }
+
+    private void checkOptionalProperty(String propertyKey, String defaultValue) {
+        this.properties.putIfAbsent(propertyKey, defaultValue);
+    }
+
+    @Override
+    public void modifyProperties(Map<String, String> properties) throws DdlException {
+        // modify properties
+        replaceIfEffectiveValue(this.properties, STORAGE_RESOURCE, properties.get(STORAGE_RESOURCE));
+        replaceIfEffectiveValue(this.properties, COOLDOWN_DATETIME, properties.get(COOLDOWN_DATETIME));
```

Comment:
We should check that only one of `COOLDOWN_DATETIME` and `COOLDOWN_TTL` can be specified